src/util/listQueryBuilder.js

```diff
@@ -520,30 +524,53 @@ async function performHistoryQuery ({
   const knex = queryBuilder
 
   const interval = intervals[groupBy]
+  const view = groupByViews[groupBy]
 
-  // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
-  // otherwise the type 'timestamp' is returned and the date won't be UTC
-  const select = knex.raw(`public.time_bucket(INTERVAL '${interval}', ??) AT TIME ZONE 'UTC' as ??`, [timeColumn, groupBy])
+  let allResults
 
-  queryBuilder = queryBuilder
-    .select(select)
-    .from(knex.raw('??.??', [schema, table]))
-    .count()
-    .groupBy(groupBy)
+  if (onlyTimeFilter && view) {
+    // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
+    // otherwise the type 'timestamp' is returned and the date won't be UTC
+    const select = knex.raw('?? AT TIME ZONE \'UTC\' as ??, count', [groupBy, groupBy])
 
-  let transformedValues
+    queryBuilder = queryBuilder
+      .select(select)
+      .from(knex.raw('??.??', [schema, view]))
 
-  if (filters) {
-    transformedValues = getTransformedValues(filters)
-    addFiltersToQueryBuilder(queryBuilder, filters, transformedValues)
-  }
+    if (filters) {
+      const newFilters = _.pick(filters, timeFilter)
+      newFilters[timeFilter].dbField = groupBy
+      const transformedValues = getTransformedValues(newFilters)
+      addFiltersToQueryBuilder(queryBuilder, newFilters, transformedValues)
+    }
 
-  const { orderBy, order } = orderConfig
-  queryBuilder.orderBy(orderBy, order)
+    const { orderBy, order } = orderConfig
+    queryBuilder.orderBy(orderBy, order)
 
-  const { page, nbResultsPerPage } = paginationConfig
+    allResults = await queryBuilder
+  } else {
+    // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
+    // otherwise the type 'timestamp' is returned and the date won't be UTC
+    const select = knex.raw(`public.time_bucket(INTERVAL '${interval}', ??) AT TIME ZONE 'UTC' as ??`, [timeColumn, groupBy])
```

Comment:
Not very DRY.
And can you explain why time_bucket is not used when `onlyTimeFilter && view` above? Already embedded in continuous aggregate? Not obvious when you read the code.

Reply:
I'll add a comment on that point.
The `time_bucket` is already embedded in continuous aggregate definition indeed.
https://github.com/stelace/stelace/blob/1306c1b13e162567e60f7f04668507061d4f1612/migrations/util/timescaleDB.js#L62-L67